tests/test_utils/tests.py

```diff
@@ -243,19 +244,44 @@ def setUpTestData(cls):
         cls.p1 = Person.objects.create(name='p1')
         cls.p2 = Person.objects.create(name='p2')
 
+    def test_empty(self):
+        self.assertQuerysetEqual(
+            Person.objects.filter(name='p3'),
+            []
+        )
+
     def test_ordered(self):
         self.assertQuerysetEqual(
             Person.objects.all().order_by('name'),
-            [repr(self.p1), repr(self.p2)]
+            [self.p1, self.p2]
         )
 
     def test_unordered(self):
         self.assertQuerysetEqual(
             Person.objects.all().order_by('name'),
-            [repr(self.p2), repr(self.p1)],
+            [self.p2, self.p1],
             ordered=False
         )
 
+    def test_queryset(self):
+        self.assertQuerysetEqual(
+            Person.objects.all().order_by('name'),
+            Person.objects.all().order_by('name')
+        )
+
+    def test_flat_values_list(self):
+        self.assertQuerysetEqual(
+            Person.objects.all().order_by('name').values_list('name', flat=True),
+            ['p1', 'p2']
+        )
+
+    @ignore_warnings(category=RemovedInDjango40Warning)
+    def test_list_of_strs(self):
+        self.assertQuerysetEqual(
+            Person.objects.all().order_by('name'),
+            [self.p1, self.p2]
+        )
+
```

Comment:
Why don't we test the deprecation message itself with a `self.assertWarnsMessage()` case?